### ISSUE TITLE
no-restricted-imports: message is ignored for star-imports when importNames is defined

### ISSUE THREAD

**Issue by mgol** [eslint/eslint#11580]
**Tell us about your environment**

* **ESLint Version:** `v5.15.3` (but the same happens on `v5.16.0` as well)
* **Node Version:** `v8.15.1`
* **~~npm~~ Yarn Version:** `1.15.2`

**What parser (default, Babel-ESLint, etc.) are you using?**

`@typescript-eslint/parser` but it's reproducible with the default one as well

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
  "parserOptions": {
    "ecmaVersion": 2018,
    "sourceType": "module",
    "rules": {
      "no-restricted-imports": [
        "error",
        {
          "paths": [
            {
              "name": "lodash-es",
              "importNames": ["anything"],
              "message": "Custom message"
            }
          ]
        }
      ],
    }
  }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
import * as _ from 'lodash-es';
```

```bash
eslint path/to/my/file.js
```

**What did you expect to happen?**

The provided message should be printed.

**What actually happened? Please include the actual, raw output from ESLint.**

The custom message is not printed, only the default one:
```
* import is invalid because 'anything' from 'lodash-es' is restricted
```

This behavior is even [validated by tests](https://github.com/eslint/eslint/blob/3396c3e2231b5b6e16da8751c22c86c256590f6b/tests/lib/rules/no-restricted-imports.js#L265-L278). As you can see, the `message` field is specified but it's not used in the output.

**Are you willing to submit a pull request to fix this bug?**

Yes

**Comment by platinumazure** [comment]
Hi @mgol, thanks for the bug report.

I can reproduce in [this demo](https://eslint.org/demo/#eyJ0ZXh0IjoiLyogZXNsaW50IFwibm8tcmVzdHJpY3RlZC1pbXBvcnRzXCI6IFtcclxuICAgICAgICBcImVycm9yXCIsXHJcbiAgICAgICAge1xyXG4gICAgICAgICAgXCJwYXRoc1wiOiBbXHJcbiAgICAgICAgICAgIHtcclxuICAgICAgICAgICAgICBcIm5hbWVcIjogXCJsb2Rhc2gtZXNcIixcclxuICAgICAgICAgICAgICBcImltcG9ydE5hbWVzXCI6IFtcImFueXRoaW5nXCJdLFxyXG4gICAgICAgICAgICAgIFwibWVzc2FnZVwiOiBcIkN1c3RvbSBtZXNzYWdlXCJcclxuICAgICAgICAgICAgfVxyXG4gICAgICAgICAgXVxyXG4gICAgICAgIH1cclxuICAgICAgXSAqL1xyXG4gICAgIFxyXG5pbXBvcnQgKiBhcyBfIGZyb20gXCJsb2Rhc2gtZXNcIjsiLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjYsInNvdXJjZVR5cGUiOiJtb2R1bGUiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnt9LCJlbnYiOnt9fX0=). Marking as accepted.

**Comment by mgol** [comment]
@platinumazure I've noticed currently you can't provide messages at all to `patterns`, only when specifying `paths`. Would a change allowing `patterns` to be defined by objects with keys `name` & `message` be accepted as well?

**Comment by kaicataldo** [comment]
@mgol Let's keep this issue scoped to the bug report. Please feel free to make another issue with a proposal to enhance the rule.

**Comment by mgol** [comment]
Will do. Thanks for the reply!
-- 
Michał Gołębiowski-Owczarek
